lib/loading.js

```diff
@@ -56,6 +57,14 @@ class LoadingUI {
     this.broadcastState()
   }
 
+  setErrorState (error) {
+    this.setStates([{
```

Comment:
What do you think to keep `error` out of states? A separate field in JSON seems to be more explicit and less hacky.

Reply:
Sure, it introduces some extra management with regards to error vs states but fine with me